MatRad_MCemittanceBaseData.m

```diff
@@ -98,42 +98,71 @@
-                        count = count + 1;
-                        continue;
-                    end
+                if isfield(machine.data(i), 'energySpectrum')
+                        if isfield(machine.data(i).energySpectrum, 'mean') && isfield(machine.data(i).energySpectrum, 'spread')
+                            energyData.NominalEnergy    = ones(1,4) * machine.data(i).energy(:);
+                            energyData.MeanEnergy       = machine.data(i).energySpectrum.mean(:);
+                            energyData.EnergySpread     = machine.data(i).energySpectrum.spread(:);
+                        else
+                            energyData = obj.fitPhaseSpaceForEnergy(i);
+                        end
+                    else
+                        energyData = obj.fitPhaseSpaceForEnergy(i);
                 end
-                
-                
-                %calculate monteCarloData for given energy and every focus
-                %index
-                data = [];
-                energyData = obj.fitPhaseSpaceForEnergy(i);
-                obj.FWHMatIso = [];
-                for j = 1:size(machine.data(i).initFocus.sigma,1)
                     
-                    tmp = energyData;
-                    opticsData = obj.fitBeamOpticsForEnergy(i, j);
+                if isfield(machine.data(i).initFocus,'Emittance') 
+                    data = [];
+                    opticsData.ProtonsMU        = ones(1,4) * 1e6;   
+                    opticsData.Weight1          = machine.data(i).initFocus.Emittance.weight.first(:);
+                    opticsData.SpotSize1x       = machine.data(i).initFocus.Emittance.spotsize.x1(:);
+                    opticsData.Divergence1x     = machine.data(i).initFocus.Emittance.divergence.x1(:);
+                    opticsData.Correlation1x    = machine.data(i).initFocus.Emittance.correlation.x1(:); 
+                    opticsData.SpotSize1y       = machine.data(i).initFocus.Emittance.spotsize.y1(:);
+                    opticsData.Divergence1y     = machine.data(i).initFocus.Emittance.divergence.y1(:);
+                    opticsData.Correlation1y    = machine.data(i).initFocus.Emittance.correlation.y1(:); 
+                    opticsData.Weight2          = machine.data(i).initFocus.Emittance.weight.second(:);
+                    opticsData.SpotSize2x       = machine.data(i).initFocus.Emittance.spotsize.x2(:);
+                    opticsData.Divergence2x     = machine.data(i).initFocus.Emittance.divergence.x2(:);
+                    opticsData.Correlation2x    = machine.data(i).initFocus.Emittance.correlation.x2(:);
+                    opticsData.SpotSize2y       = machine.data(i).initFocus.Emittance.spotsize.y2(:);
+                    opticsData.Divergence2y     = machine.data(i).initFocus.Emittance.divergence.y2(:);
+                    opticsData.Correlation2y    = machine.data(i).initFocus.Emittance.spotsize.y2(:);
                     
+                    tmp = energyData;
```

Comment:
I don't really understand this & the following loop. Is there no other way to concatenate the structs?